### PR TITLE
Feature/75 filter

### PR DESCRIPTION
**Pull request by amazeika**

### CODE REVIEW

**Comment by amazeika** [comment]
@ercanozkaya Can you please review this man?

File: model/state/nodes.php [hunk 1]
```diff
@@ -19,7 +19,7 @@ public function __construct(KObjectConfig $config)
     {
         parent::__construct($config);
 
-        $this->insert('uri', 'url');
+        $this->insert('uri', 'string');
```
Comment (ercanozkaya) [inline model/state/nodes.php]: @amazeika string filter will change & to &amp; for example. do we want that?

File: model/state/nodes.php [hunk 2]
```diff
@@ -30,9 +30,16 @@ public function set($name, $value = null)
 
             if (!isset($parts['scheme']) || $parts['scheme'] == 'file')
             {
-                $this->set('name', basename($parts['path']));
+                $path = $parts['path'];
 
-                $folder = dirname($parts['path']);
+                // Handle folders containing question marks
+                if (isset($parts['query'])) {
```
Comment (ercanozkaya) [inline model/state/nodes.php]: @amazeika do we need to do the same for fragment if # character is used?

File: model/state/thumbnails.php [hunk 3]
```diff
@@ -22,7 +22,7 @@ public function __construct(KObjectConfig $config)
         parent::__construct($config);
 
         $this->insert('version', 'cmd')
-             ->insert('source', 'url');
+             ->insert('source', 'string');
```
Comment (ercanozkaya) [inline model/state/thumbnails.php]: @amazeika same question as above

File: model/state/thumbnails.php [hunk 4]
```diff
@@ -35,8 +35,15 @@ public function set($name, $value = null)
 
             $parts = parse_url($value);
 
-            $this->set('name', basename($parts['path']) . '.jpg');
-            $this->set('folder', trim(dirname($parts['path']), '/'));
+            $path = $parts['path'];
+
+            // Handle folders containing question marks
+            if (!isset($parts['scheme']) || $parts['scheme'] == 'file' && isset($parts['query'])) {
+                $path .= sprintf('?%s', $path);
```
Comment (ercanozkaya) [inline model/state/thumbnails.php]: @amazeika same as above

**Pull request merged**